### DISCUSSION TITLE
Typespec integration ( https://typespec.io )

### DISCUSSION THREAD

**Discussion by creatorrr** [scalar/scalar#3039] [Ideas]
Props for the amazing work on scalar. It'd be great to have an integration with [TypeSpec](https://typespec.io/) -- a microsoft-backed DSL for defining APIs. We use it extensively and having a [typespec library](https://typespec.io/docs/extending-typespec/basics) for scalar (esp, the custom extensions and annotations) would be dope.

**Comment by hanspagel** [comment]
That sounds interesting. Would you be able to work on such a library and submit a PR?

**Reply by creatorrr** [reply]
Not at the moment unfortunately. But will def try to soon
